src/arch/main.cpp

```diff
@@ -52,27 +38,35 @@ int main(int argc, char *arg[]){
 
   checkEndianess();
 
-  startupHook(argc, arg);
+  if(auto result = CForteArchitecture::initialize(argc, arg); result != 0){
+     return result;
+  }
 
-#ifdef FORTE_IO_PLCNEXT
-  sleep(3);
-  DeviceStatus::startup();
-#endif
+  struct callOnExit {
```

Comment:
Open for discussion, but do we really want to use structs as more than plain data containers? @azoitl @mx990 
From my perspective, its would be clearer to use classes everywhere we need more than a collection of data entries, and reserve the use of structs exactly for that, plain data containers.
My rationale for this is, that you know what to expect if you see class or struct that way, without needing to check the code.
The cost is quite low, only adding `public:` as additional line here

Reply:
Why not change `CForteArchitecture::deinitialize` to a destructor and use a local variable of type `CForteArchitecture` to avoid the additional struct? Alternatively, there is [std::atexit](https://en.cppreference.com/w/cpp/utility/program/atexit).

Reply:
I like the `std::atexit` since it applies in the context of the `main` function in a c++ environment.